### DISCUSSION TITLE
My ip address in the result

### DISCUSSION THREAD

**Discussion by gauravdash** [benbusby/whoogle-search#664] [Q&A] [answered]
Hi,

I run whoogle on a dedicated server machine which has its own public ip address.
When I search for "my ip address" I see my public ip address in the top results. This is the ip address of the machine running the browser.
I was expecting ip address of the server would appear in the results.

How does google know where my browser is running?

I do not see any javascript in results page that is connecting back to google to get my browser's public ip address.

Any ideas?

Thanks.

**Comment by Albonycal** [comment]
it used to show the IP address of the server, but this's a new feature now,
it shows the client IP.
It doesn't log it and you can look at the source code.

**Reply by gauravdash** [reply]
Hi,
Thank for you reply.

I understand that it doesn't log it.
But the point I want to make is that the client ip (browser ip) is sent to google.
Or google is able to get that in some way which means whoogle proxy needs to be fortified to not let client ip leak through.

First of all,
To change anything in whoogle or even to decide whether a change is required or not,
it would help to understand how google is able to get the client ip address.

Thanks.

**Comment by Albonycal** [comment, marked as answer]
no it's not the client IP isn't sent to google
that thing is done locally by whoogle
i.e adding the client IP

**Reply by gauravdash** [reply]
Hi,

Ok, I see what you mean.
I though the result from google is relayed.
But instead whoogle generates the ip address result card.
Code is here - app/utils/results.py

Thanks.
